controllers/numaresourcesoperator_controller.go

```diff
@@ -462,28 +474,28 @@ func (r *NUMAResourcesOperatorReconciler) syncNUMAResourcesOperatorResources(ctx
 	}
 
 	var err error
-	dssByMCP := make(map[string]nropv1.NamespacedName)
-
+	// using a slice of MCPDaemonSetPair instead of a map because Go maps assignment order is not consistent and indeterministic
```

Comment:
no need to reiterate, the rationale is already well explained in the commit message. But not blocking either.

Reply:
I thought it would be handy here to avoid the thoughts of "this scenario fits into a map so let's change this to use map". I can adjust the comment to focus more on what slice provides and its importance, or completely remove that. what do you see better fit?

Reply:
right, good point. Let's keep it.